core/src/main/scala/algebra/Order.scala

```diff
@@ -95,6 +95,21 @@ trait Order[@sp A] extends Any with PartialOrder[A] { self =>
    */
   override def gt(x: A, y: A): Boolean =
     compare(x, y) > 0
+
+  /**
+   * Returns a new `Order[A]` instance that first compares by the original
+   * `Order` instance and uses the provided `Order` instance to "break ties".
+   *
+   * That is, `x.combine(y)` creates an `Order` that first orders by `x` and
+   * then (if two elements are equal) falls back to `y` for the comparison.
+   */
+  def andThen(o: Order[A]): Order[A] = new Order[A] {
```

Comment:
When I named it "andThen" I was thinking that when I talk about ordering I might say something like "order by X and then Y and then Z". But I'm not attached to this name. I'll change it to `whenEqual`.